extensions/big-data-cluster/src/wizards/create-cluster/pages/settingsPage.ts

```diff
@@ -32,15 +32,25 @@ export class SettingsPage extends WizardPageBase<CreateClusterModel> {
 		let dockerSettingsGroup = view.modelBuilder.groupContainer().withItems([]).withLayout({ header: localize('bdc-create.DockerSettingsText', 'Docker Settings'), collapsible: true }).component();
 
 		let acceptEulaCheckbox = view.modelBuilder.checkBox().component();
-		acceptEulaCheckbox.label = localize('bdc-create.AcceptEulaText', 'I accept the SQL Server EULA');
 		acceptEulaCheckbox.checked = false;
 
-		let eulaHyperlink = view.modelBuilder.hyperlink().withProperties({
-			label: localize('bdc-create.ViewEulaText', 'View Eula'),
+		let eulaLink: sqlops.LinkArea = {
+			startPosition: parseInt(localize('bdc-create.EulaLinkAreaStartPosition', '13')),
```

Comment:
Why are we localizing integer values? I don't think shows the right solution, are you suggesting overriding the int values on a per-locale basis? I think we're better off just defining a LinkArea as a a set of string+link values, e.g.
```
{
   text: localize('bdc-create.eula', 'License Agreement'),
   url: 'https://docs.microsoft.com/en-us/sql/getting-started/about-the-sql-server-license-terms?view=sql-server-2014'
}
```

Then you can just do DOM building to make this all work. It's a cleaner overall solution that's robust to localization

Reply:
This solution won't work when we have multiple occurrences of a substring, but you don't want to add link for all of them. I leveraged the design from WinForms LinkLabel control:  https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.linklabel.linkarea?view=netframework-4.7.2

Reply:
@kevcunnane , I found that localize function allows us to pass in comments for the translator, we can provide comment so that they will know what needs to be updated together with the string update. :) Does it address some of your concerns?

Reply:
Do the tools that we use to localize strings make it easy to figure out which string the index would be referring to and to see the localized version of that string while figuring out the value for the index?

With Kevin's approach the overall string would be something like `localize('I accept the {0} and {1}')`, and the other strings would be `localize('License Agreement')` and `localize('Privacy Policy')`. Then we can replace `{0}` and `{1}` where they appear in text

Reply:
It partially helps, but I would still check with localization.

As general feedback I would still push back on this solution because it's more cumbersome to localizers and extension point users to figure out the indexes and track across languages. I get that Winforms does this but that is based on having UI-driven development that sets things automatically. WPF/XAML and other declarative frameworks seem to avoid the magic numbers approach more often. It is more work for us to build, but then easier in the long run to build on top of.